server/snapshot/snapshot.go

```diff
@@ -16,35 +16,53 @@ import (
 // GetOrCreateSnapshotKey either creates a new snapshot key, or returns
 // the existing one. Only the PublicKey is returned. The private part
 // is held by the CryptoService.
-func GetOrCreateSnapshotKey(gun string, store storage.KeyStore, crypto signed.CryptoService, createAlgorithm string) (data.PublicKey, error) {
-	keyAlgorithm, public, err := store.GetKey(gun, data.CanonicalSnapshotRole)
-	if err == nil {
-		return data.NewPublicKey(keyAlgorithm, public), nil
-	}
-
-	if _, ok := err.(*storage.ErrNoKey); ok {
+func GetOrCreateSnapshotKey(gun string, store storage.MetaStore, crypto signed.CryptoService, createAlgorithm string) (data.PublicKey, error) {
+	_, rootJSON, err := store.GetCurrent(gun, data.CanonicalRootRole)
+	if err != nil {
+		// If the error indicates we couldn't find the root, create a new key
+		if _, ok := err.(storage.ErrNotFound); !ok {
+			logrus.Errorf("Error when retrieving root role for GUN %s: %v", gun, err)
+			return nil, err
+		}
 		key, err := crypto.Create(data.CanonicalSnapshotRole, gun, createAlgorithm)
```

Comment:
group review:  we can just `return crypto.Create(data.CanonicalSnapshotRole, gun, createAlgorithm)` here - similarly for `GetOrCreateTimestampKey`